docker_ci_deploy/__main__.py

```diff
@@ -9,25 +9,82 @@
 from itertools import chain
 
 
-def strip_image_tag(image):
+def split_image_tag(image_tag):
     """
-    Remove the tag part from a Docker tag and return the rest. Full tags are of
-    the form [REGISTRYHOST/][NAME/...]NAME[:TAG] where the REGISTRYHOST may
-    contain a ':' but no '/', the NAME parts may contain '/' but no ':', and
-    the TAG part may contain neither ':' nor '/'.
+    Split the given image tag into its name and tag parts (<name>[:<tag>]).
+
+    Full tags are of the form [REGISTRYHOST/][NAME/...]NAME[:TAG] where the
+    REGISTRYHOST may contain a ':' but no '/', the NAME parts may contain '/'
+    but no ':', and the TAG part may contain neither ':' nor '/'.
     """
-    match = re.match('^((?:[^\/]+\/)?[^:]+)(?::[^:\/]+)?$', image)
+    match = re.match(r'^((?:[^\/]+\/)?[^:]+)(?::([^:\/]+))?$', image_tag)
```

Comment:
Maybe I need to redo this regex and break it into different parts.. but that would probably be a different PR.